runtime/sema/type.go

```diff
@@ -2533,6 +2569,16 @@ func ArraySliceFunctionType(elementType Type) *FunctionType {
 	)
 }
 
+func ArrayToVariableSizedFunctionType(elementType Type) *FunctionType {
+	return NewSimpleFunctionType(
+		FunctionPurityImpure,
+		[]Parameter{},
```

Comment:
Can this be `nil`?

Reply:
I think all those usages are nil-safe. For example, `t.Parameters` is always used in for-loops or with `len(t.Parameters)`, both of which handle `nil` gracefully.
There are already function types with `nil` parameters. e.g:
https://github.com/onflow/cadence/blob/08be35302d32825a01001d8ba217fd32d15e8f51/runtime/sema/runtime_type_constructors.go#L27-L31

```suggestion
		nil,
```